pandas/util/testing.py

```diff
@@ -1118,10 +1118,12 @@ def assert_extension_array_equal(left, right):
     right_na = right.isna()
     assert_numpy_array_equal(left_na, right_na)
 
-    left_valid = left[~left_na].astype(object)
-    right_valid = right[~right_na].astype(object)
+    if len(left_na) > 0 and len(right_na) > 0:
```

Comment:
If `left` has length zero, then the statement that follows, i.e.
```
left_valid = left[~left_na].astype(object)
```
fails.  There was some test case that was failing without this change (but I don't remember which one, and it may have been in the ops stuff).  I will leave this change out and see if this fix for `combine` still works.

Reply:
Given that this works

```
In [9]: pd.Series()[np.array([], dtype='bool')].astype(object)
Out[9]: Series([], dtype: object)
```

then we should ensure that Series[extension_array] works too. If you find a reproducible example, could you make a new issue?

Reply:
I took this code out of this PR.  It's not needed here.  When we get to the ops stuff, I'll have to reintroduce it, and figure out the boundary case that required this particular case.